.github/workflows/analysis.yml

```diff
@@ -0,0 +1,157 @@
+        id: cef-cache
+        uses: actions/cache@v3
+        env:
+          CACHE_NAME: 'cef-cache'
+        with:
+          path: ${{ github.workspace }}/obs-build-dependencies/cef_binary_${{ env.CEF_BUILD_VERSION_LINUX }}_linux64
+          key: ${{ runner.os }}-pr-${{ env.CACHE_NAME }}-${{ env.CEF_BUILD_VERSION_LINUX }}-${{ env.CACHE_REVISION }}
+
+      - name: 'Setup PVS Studio'
+        run: |
+          wget -q -O - https://files.pvs-studio.com/etc/pubkey.txt | sudo apt-key add -
+          sudo wget -O /etc/apt/sources.list.d/viva64.list https://files.pvs-studio.com/etc/viva64.list
+          sudo apt update
+          sudo apt install pvs-studio
+          pvs-studio-analyzer credentials ${{ secrets.PVS_NAME }} ${{ secrets.PVS_KEY }}
+
+      - name: 'Install dependencies'
+        env:
+          RESTORED_CEF: ${{ steps.cef-cache.outputs.cache-hit }}
+        run: CI/linux/01_install_dependencies.sh --disable-pipewire
+
+      - name: 'Build OBS'
+        run: CI/linux/02_build_obs.sh --disable-pipewire
+
+      - name: 'Upload PVS Logs'
+        uses: actions/upload-artifact@v3
+        with:
+          name: 'pvs-logs-linux'
+          path: 'obs-studio/build/pvs-results.sarif'
+
+      - name: 'Upload PVS report'
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: 'obs-studio/build/pvs-results.sarif'
+          checkout_path: 'obs-studio'
+          category: 'PVS-Studio (Linux)'
+
+  pvs_windows:
+    name: 'PVS Studio (Windows)'
+    runs-on: 'windows-2022'
+    env:
+      CMAKE_GENERATOR: 'Visual Studio 17 2022'
+      CMAKE_SYSTEM_VERSION: '10.0.18363.657'
+      VIRTUALCAM-GUID: 'A3FCE0F5-3493-419F-958A-ABA1250EC20B'
+    defaults:
+      run:
+        working-directory: 'obs-studio'
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+          path: 'obs-studio'
+          fetch-depth: 0
+
+      - name: 'Add msbuild to PATH'
+        uses: microsoft/setup-msbuild@v1.1
+
+      - name: 'Restore VLC dependency from cache'
+        id: vlc-cache
+        uses: actions/cache@v3
+        env:
+          CACHE_NAME: 'vlc-cache'
+        with:
+          path: ${{ github.workspace }}/obs-build-dependencies/vlc-${{ env.VLC_VERSION_WIN }}
+          key: ${{ runner.os }}-pr-${{ env.CACHE_NAME }}-${{ env.VLC_VERSION_WIN }}-${{ env.CACHE_REVISION }}
+
+      - name: 'Restore Chromium Embedded Framework from cache'
+        id: cef-cache
+        uses: actions/cache@v3
+        env:
+          CACHE_NAME: 'cef-cache'
+        with:
+          path: ${{ github.workspace }}/obs-build-dependencies/cef_binary_${{ env.CEF_BUILD_VERSION_WIN }}_windows_x64
+          key: ${{ runner.os }}-pr-${{ env.CACHE_NAME }}-${{ env.CEF_BUILD_VERSION_WIN }}-x64-${{ env.CACHE_REVISION }}
+
+      - name: 'Setup PVS Studio'
+        run: |
+          New-Item -Path ..\PVSTestResults -ItemType Directory
```

Comment:
Good enough for now - for the next revision of the build scripts I'd add this to the dependency resolver based on build configuration.